utils/github/local.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+
+# `pip install …`
+import git # gitpython
+
+import functools
+import os
+import re
+
+
+class Local:
+    RE_STABLE_REF = re.compile(r'^refs/remotes/.+/\d+\.\d+$')
+
+    def __init__(self, repo_path, remote_name, default_branch_name):
+        self._repo = git.Repo(repo_path, search_parent_directories=(not repo_path))
+        self._remote = self._repo.remotes[remote_name]
+        self._default = self._remote.refs[default_branch_name]
+
+        # public key comparator
+        def cmp(x, y):
+            if x == y:
+                return 0
+            if self._repo.is_ancestor(x, y):
+                return -1
+            else:
+                return 1
+        self.comparator = functools.cmp_to_key(cmp)
```

Comment:
Why so complex? Can't we make it normal function with two parameters? `x` less than than `y` iff `x != y and not self._repo.is_ancestor(x, y)`?

Reply:
The `cmp_to_key()` is a Python 3 interface adapter for simple `cmp`-like functions: https://portingguide.readthedocs.io/en/latest/comparisons.html#rich-comparisons

Reply:
Ok, but why we use it? Simple function is much more readable.

Reply:
We use it inside `sorted()` to sort the stable branches' bases - and thus sort the branches.